### PR TITLE
Update CircleCI benchmarks with new env variables

### PR DESCRIPTION
**Pull request by duffn**
## Description

- Update the CircleCI benchmarks to use necessary environment variables for benchmark runs
  - `BENCHMARK_CONTENT_SOURCE`
  - `BENCHMARK_REPO_NAME`
  - `BENCHMARK_SITE_TYPE`
- Extract the benchmark run into a command
- Note a specific cold vs. warm environment variable in `BENCHMARK_BUILD_TYPE` for subsequent runs

I don't know how to test this. We wait and see if this works?

cc/ @nicholascapo

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -365,193 +408,145 @@ jobs:
...
-      - run:
-          command: npm install
-          working_directory: benchmarks/markdown_id
-      - run:
-          command: npm run build
+      - run-benchmark:
           working_directory: benchmarks/markdown_id
+          NUM_PAGES: "512"
+          BENCHMARK_CONTENT_SOURCE: MDX
+          BENCHMARK_REPO_NAME: gatsbyjs/gatsby
```
[the remark this answers is not in the document]
Comment (duffn) [reply .circleci/config.yml]: I agree. I've updated the repo names.

**Comment by duffn** [comment]
As for testing, I'm seeing if I can run this on my fork.

**Comment by duffn** [comment]
I forced a couple of these to run on my fork. They completed successfully and passed the correct environment variables to the service. https://app.circleci.com/pipelines/github/duffn/gatsby/7/workflows/539c1ff6-fdfa-4ad8-9400-d77d506d7c37/jobs/56

File: .circleci/config.yml [hunk 2]
```diff
@@ -130,6 +130,49 @@ commands:
...
+          environment:
+            BENCHMARK_BUILD_TYPE: COLD_START
+            NUM_PAGES: << parameters.NUM_PAGES >>
+            BENCHMARK_CONTENT_SOURCE: << parameters.BENCHMARK_CONTENT_SOURCE >>
+            BENCHMARK_REPO_NAME: << parameters.BENCHMARK_REPO_NAME >>
+            BENCHMARK_SITE_TYPE: << parameters.BENCHMARK_SITE_TYPE >>
+            CI_NAME: circleci
+      - run:
+          command: npm install
+          working_directory: << parameters.working_directory >>
```
Comment (pieh) [inline .circleci/config.yml]: is deps installation needed after first run?

File: .circleci/config.yml [hunk 3]
```diff
@@ -365,193 +408,145 @@ jobs:
...
-      - checkout
-      - run:
-          command: npm install
-          working_directory: benchmarks/markdown_id
-      - run:
-          command: npm run build
+      - run-benchmark:
           working_directory: benchmarks/markdown_id
+          NUM_PAGES: "512"
+          BENCHMARK_CONTENT_SOURCE: MDX
```
Comment (pieh) [inline .circleci/config.yml]: I'm not sure if there is distinction needed - but technically this benchmark is `MD` and not `MDX` (and similarly all others)

Comment (duffn) [reply .circleci/config.yml]: This relates to an enumeration that we have in cloud where the values is `MDX`.

File: .circleci/config.yml (hunk 2 above)
Comment (duffn) [reply .circleci/config.yml]: Probably not technically, but I've left it in to be consistent with all of the other benchmarks we have setup. 

If it causes any issues, we can remove it everywhere.

Comment (pieh) [reply .circleci/config.yml]: It's fine - just was checking of not lost copy pasta ;)

**Pull request merged**